utils/land-patterns.stanza

```diff
@@ -5,281 +5,406 @@ defpackage ocdb/land-patterns :
-  type = SMD
-  shape = Rectangle(anchor, w, h)
-  layer(Paste(Top)) = Rectangle(anchor, w, h)
-  layer(SolderMask(Top)) = Rectangle(anchor, w, h)
-
-public pcb-pad bga-pad (d:Double) :
-  name = to-string("%_ BGA Pad" % [d])
-  type = SMD
-  shape = Circle(d / 2.0)
-  layer(Paste(Top)) = Circle(d / 2.0)
-  layer(SolderMask(Top)) = Circle(d / 2.0)
-
-public pcb-pad top-smd-testpoint-pad (d:Double) :
-  name = to-string("Test Point Pad")
-  type = SMD
-  shape = Circle(d / 2.0)
-  layer(Paste(Top)) = Circle(d / 2.0)
-  layer(SolderMask(Top)) = Circle(d / 2.0)
-
-public pcb-pad btm-smd-testpoint-pad (d:Double) :
-  name = to-string("Test Point Pad")
-  type = SMD
-  shape = Circle(d / 2.0)
-  layer(Paste(Bottom)) = Circle(d / 2.0)
-  layer(SolderMask(Bottom)) = Circle(d / 2.0)
-
-public defn top-smd-testpoint-pad () :
-  top-smd-testpoint-pad(2.0)
-
-public defn btm-smd-testpoint-pad () :
-  btm-smd-testpoint-pad(2.0)
-
-public pcb-pad testpoint-pad (d:Double) :
-  name = to-string("%_  Testpoint Pad" % [d])
-  type = SMD
-  shape = Circle(d / 2.0)
-  layer(SolderMask(Top)) = Circle(d / 2.0)
-    
-public defn smd-pad (w:Double, h:Double) :
-  smd-pad(C,w,h)
-
-pcb-pad pth-pad (anchor:Anchor, drill-r:Double, pad-r:Double, mask-r:Double) :
-  name = to-string("%_ PTH Pad (drill = %_, pad = %_, mask = %_)" % [
-           anchor, drill-r, pad-r, mask-r])
-  type = TH
-  shape = Circle(anchor, pad-r)
-  layer(Cutout()) = Circle(anchor, drill-r)
-  layer(SolderMask(Top)) = Circle(anchor, mask-r)
-  layer(SolderMask(Bottom)) = Circle(anchor, mask-r)
-
-public defn pth-pad (drill-r:Double, mask-r:Double) :
-  pth-pad(C, drill-r, mask-r, mask-r)
-
-public defn pth-pad (anchor:Anchor, drill-r:Double) :
-  pth-pad(anchor, drill-r, drill-r + 0.35, drill-r + 0.35)
-
-public defn pth-pad (drill-r:Double) :
-  pth-pad(C, drill-r)
+  ref-label(loc(0.0, 0.0))
+
+; Create a capsule with normalized rotation
+defn capsule (anchor:Anchor, width:Double, height:Double) -> Capsule:
+  if width >= height:
+    Capsule(anchor, width, height)
+  else:
+    Capsule(width, height, loc(0.0, 0.0, 90.0))
+
+; Create a rounded rectangle with normalized rotation
+defn round-rect (width:Double, height:Double, r:Double) -> RoundedRectangle:
+  if width >= height:
+    RoundedRectangle(width, height, r)
+  else: 
+    RoundedRectangle(height, width, r, loc(0.0, 0.0, 90.0))
+
+; Create a chamfered rectangle with normalized rotation
+defn chamfered (width:Double, height:Double, r:Double) -> ChamferedRectangle:
+  if width >= height:
+    ChamferedRectangle(width, height, r)
+  else:
+    ChamferedRectangle(height, width, r)
+
+;==============================================================================
+;=========================== Geometric Helpers ================================
+;==============================================================================
+; Create a rectangle out of lines
+public defn LineRectangle (w:Double, h:Double, xc:Double, yc:Double, line-w:Double) :
```

Comment:
Please do a grep through the most up-to-date version of the codebase. I believe Bertrand might have already relocated these `LineRectangle` helpers to a much more sensible location. If so, then you can remove these definitions.

Reply:
That is a major blocker, since we cannot rebuild a test release yet. OCDB has to be tested against the most recent stable release and cannot refer to code that isn't in the stable client that we test against.

Reply:
Okay. Let's merge it and stage it then. 

This part of CI needs to be figured out.